.github/workflows/deploy-frontend.yml

```diff
@@ -4,28 +4,37 @@ on:
   workflow_call:
     inputs:
       report_artifact_in_scope:
-        required: false
         type: boolean
+        default: false
   workflow_dispatch:
   release:
     types: [published]
 
 jobs:
+  # if called as a separate workflow
   build-frontend:
     uses: ./.github/workflows/build-frontend.yml
+    if: inputs.report_artifact_in_scope == false
+
   collect:
     runs-on: ubuntu-latest
+    if: ${{ !failure() }}
     needs: build-frontend
     steps:
       - name: Create _site directory
         run: mkdir _site
-      - name: Download frontend build
-        uses: actions/download-artifact@v4
+
+      # if called from report workflow
+      - name: Download latest frontend-build
+        uses: dawidd6/action-download-artifact@v3
+        if: inputs.report_artifact_in_scope == true
         with:
+          workflow: build-frontend.yml
+          branch: main
           name: frontend-build
           path: _site
+          if_no_artifact_found: warn
 
```

Comment:
I am most concerned about this. Is this the correct way of downloading whatever last frontend-build was available ?

Reply:
It seems to be downloading it correctly from the ss above.